AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/releases/ReleasesRest.java

```diff
@@ -47,19 +52,95 @@ public class ReleasesRest {
 
     @GET
     @ApiOperation(value = "Get releases", notes = "Returns all releases")
-    public List<ReleaseEntity> getReleases() {
-        return releaseMgmtService.loadAllReleases(true);
+    public List<ReleaseEntity> getReleases(@DefaultValue("-1") @QueryParam("start") int start, @DefaultValue("-1") @QueryParam("limit") int limit) {
+        if (start < 0 || limit < 0) {
+            return releaseLocator.loadAllReleases(true);
+        } else {
+            return releaseLocator.loadReleasesForMgmt(start, limit, true);
+        }
     }
 
     @GET
     @Path("/{id}")
     @ApiOperation(value = "Get a release", notes = "Returns the specifed release")
     public Response getRelease(@PathParam("id") int id) {
         ReleaseEntity release = releaseLocator.getReleaseById(id);
```

Comment:
You could move the "logic" into a UseCase - that would then throw a NotFoundException, if no release was found. The NotFoundException is then mapped to the proper response. See `ch.mobi.itc.mobiliar.rest.exceptions.NotFoundExceptionMapper`
This way, the Jax-RS Endpoint implementation become as clean as possible...